packages/grid/_modules_/grid/models/gridComponentOverridesProp.tsx

```diff
@@ -29,4 +30,8 @@ export interface GridComponentOverridesProp {
    * Error overlay component rendered above the grid when an error is caught.
    */
   errorOverlay?: React.ElementType<ComponentProps>;
+  /**
+   * ColumnMenu component rendered by clicking on the 3 dots icon in column headers.
+   */
+  columnMenu?: React.ElementType<ComponentProps & GridColumnHeaderMenuItemProps>;
```

Comment:
Should we start the capitalization API now to minimize breaking changes at the cost of temporary inconsistency or wait for #840?
```suggestion
  ColumnMenu?: React.ElementType<ComponentProps & GridColumnHeaderMenuItemProps>;
```

Reply:
Same question to anticipate and minimize BCs with the props:

```suggestion
  columnMenu?: React.ElementType<GridColumnHeaderMenuItemProps>;
```

Reply:
> Should we start the capitalization API now to minimize breaking changes at the cost of temporary inconsistency or wait for #840?

It's in my second PR, I'm trying to do small PR here :)

Reply:
OK sounds good. So we stick to consistency at the data grid level in this iteration, and work on the consistency of the UI kit level in a second run.